src/sirhelpers.c

```diff
@@ -314,25 +314,24 @@ int _sir_fopen(FILE* restrict* restrict streamptr, const char* restrict filename
 
 struct tm* _sir_localtime(const time_t* restrict timer, struct tm* restrict buf) {
     if (_sir_validptr(timer) && _sir_validptr(buf)) {
-#if defined(__HAVE_STDC_SECURE_OR_EXT1__)
+#if defined(__HAVE_STDC_SECURE_OR_EXT1___)
 # if defined(__WIN__)
         errno_t ret = (errno_t)localtime_s(buf, timer);
+# else
+        errno_t ret = (errno_t)localtime_s(timer, buf);
+# endif
         if (0 != ret) {
             (void)_sir_handleerr(ret);
             return NULL;
         }
 
         return buf;
+#else /* !__HAVE_STDC_SECURE_OR_EXT1___ */
+# if !defined(__WIN__) || defined(__EMBARCADEROC__)
+        struct tm* ret = localtime_r(timer, buf);
 # else /* __WIN__ */
-        struct tm* ret = localtime_s(timer, buf);
```

Comment:
Yeah this is wrong, and shouldn't even have compiled on windows... see the link to the docs in above comment

Reply:
On my cell but I'll fix - I'm sure I'll mess it up the first time, :)